In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [42]:
# import necessary modules
import pandas as pd
import os
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as st
import scipy

In [30]:
# load the data
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/train.csv')
df.head(3)

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  

[3 rows x 81 columns]

In [31]:
ndf = df.fillna(0)

In [63]:
dfFilter = ndf.groupby('YrSold')['SalePrice'].mean().to_frame('SalePrice').reset_index()
dfFilter

YrSold      SalePrice
0    2006  182549.458599
1    2007  186063.151976
2    2008  177360.838816
3    2009  179432.103550
4    2010  177393.674286

apakah YrSold berpengaruh terhadap harga jual?

# Penentuan Hipotesis

uji hipotesis sisi kiri

H0 : harga rumah <= harga tahun sebelumnya

H1 : harga rumah > harga tahun sebelumnya

α = 5% = 0.05


In [68]:
y2006 = ndf[ndf['YrSold']==2006]
y2007 = ndf[ndf['YrSold']==2007]
y2008 = ndf[ndf['YrSold']==2008]
y2009 = ndf[ndf['YrSold']==2009]
y2010 = ndf[ndf['YrSold']==2010]

In [69]:
anova_test = st.f_oneway(y2006['SalePrice'],y2007['SalePrice'],y2008['SalePrice'],y2009['SalePrice'],y2010['SalePrice'])

In [70]:
anova_test.pvalue

0.6300887663573457

In [71]:
if anova_test.pvalue>0.05:
    print('Ho diterma')
else:
    print('Ho ditolak')

Ho diterma


kesimpulan : ho diterima, sehingga harga rumah tidak selalu naik tiap tahunnya.

In [76]:
dfBank = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/bank-full.csv', sep=';')
dfBank.head(3)

age           job  marital  education default  balance housing loan  \
0   58    management  married   tertiary      no     2143     yes   no   
1   44    technician   single  secondary      no       29     yes   no   
2   33  entrepreneur  married  secondary      no        2     yes  yes   

   contact  day month  duration  campaign  pdays  previous poutcome   y  
0  unknown    5   may       261         1     -1         0  unknown  no  
1  unknown    5   may       151         1     -1         0  unknown  no  
2  unknown    5   may        76         1     -1         0  unknown  no

In [74]:
dfBank.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521 entries, 0 to 4520
Data columns (total 17 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        4521 non-null   int64 
 1   job        4521 non-null   object
 2   marital    4521 non-null   object
 3   education  4521 non-null   object
 4   default    4521 non-null   object
 5   balance    4521 non-null   int64 
 6   housing    4521 non-null   object
 7   loan       4521 non-null   object
 8   contact    4521 non-null   object
 9   day        4521 non-null   int64 
 10  month      4521 non-null   object
 11  duration   4521 non-null   int64 
 12  campaign   4521 non-null   int64 
 13  pdays      4521 non-null   int64 
 14  previous   4521 non-null   int64 
 15  poutcome   4521 non-null   object
 16  y          4521 non-null   object
dtypes: int64(7), object(10)
memory usage: 600.6+ KB


In [75]:
dfBank.describe()

age       balance          day     duration     campaign  \
count  4521.000000   4521.000000  4521.000000  4521.000000  4521.000000   
mean     41.170095   1422.657819    15.915284   263.961292     2.793630   
std      10.576211   3009.638142     8.247667   259.856633     3.109807   
min      19.000000  -3313.000000     1.000000     4.000000     1.000000   
25%      33.000000     69.000000     9.000000   104.000000     1.000000   
50%      39.000000    444.000000    16.000000   185.000000     2.000000   
75%      49.000000   1480.000000    21.000000   329.000000     3.000000   
max      87.000000  71188.000000    31.000000  3025.000000    50.000000   

             pdays     previous  
count  4521.000000  4521.000000  
mean     39.766645     0.542579  
std     100.121124     1.693562  
min      -1.000000     0.000000  
25%      -1.000000     0.000000  
50%      -1.000000     0.000000  
75%      -1.000000     0.000000  
max     871.000000    25.000000

In [88]:
dfBankFilter = dfBank[['campaign', 'y']]
dfBankFilter.fillna(0)
dfBankFilter


campaign    y
0             1   no
1             1   no
2             1   no
3             1   no
4             1   no
...         ...  ...
45206         3  yes
45207         2  yes
45208         5  yes
45209         4   no
45210         2   no

[45211 rows x 2 columns]

Apakah jumlah campaign mampu meningkatkan keinginan user untuk membuka akun deposit atau tidak

a. Apa nama experimentnya? menentukan hubungan antara jumlah campaign dengan keinginan user untuk deposit    

b. Definisikan Hipotesis ?
null_hypothesis = "tidak ada hubungan antara jumlah campaign dengan keinginan user untuk deposit"
alternate_hypothesis = "ada hubungan antara jumlah campaign dengan keinginan user untuk deposit"

c. Siapa participant-nya? user bank

d. Variabel yang akan diuji apa? campaign

e. Metrics apa yang akan digunakan? jumlah campaign

f. Berapa sample size dan durasi experiment? 45211,  0.030620574951171875 detik


In [86]:
from scipy.stats import chi2_contingency, chi2

In [94]:
import time
start_time = time.time()

observed_values = pd.crosstab(dfBankFilter["campaign"], dfBankFilter["y"]).values

# calculate expected frequencies & chi square statistic
chi2_statistic, p_value, dof, expected_values = chi2_contingency(observed_values, correction = False)
print(chi2_statistic, p_value)

# find the critical value for our test
critical_value = chi2.ppf(1 - 0.05, dof)
print(critical_value)

if p_value >0.05:
    print('Ho diterma')
else:
    print('Ho ditolak')

print("--- %s seconds ---" % (time.time() - start_time))    

359.4924794027183 9.896132606234023e-50
64.00111197221803
Ho ditolak
--- 0.025902509689331055 seconds ---


Pengujian Hipotesis
a. Penentuan hipotesis
null_hypothesis = "tidak ada hubungan antara jumlah campaign dengan keinginan user untuk deposit" alternate_hypothesis = "ada hubungan antara jumlah campaign dengan keinginan user untuk deposit"

b. Pengujian yang digunakan : Chi-Square Test

c. Hasil pengujian hipotesis : Ho ditolak

d. Kesimpulan dan interpretasi : ada hubungan antara jumlah campaign dengan keinginan user untuk deposit